src/Controller/Admin/UserController.php

```diff
@@ -0,0 +1,171 @@
+        string $defaultPerPage
+    ) {
+        $this->userRepository = $userRepository;
+        $this->entityManager = $entityManager;
+        $this->userPasswordHasher = $userPasswordHasher;
+        $this->defaultPerPage = (int) $defaultPerPage;
+    }
+
+    /**
+     * @Route("admin/users", name="admin_users", methods={"GET"})
+     */
+    public function getAllUsers(Request $request): Response
+    {
+        $paginate['currentPage'] = $request->query->get('page', 1);
+        $paginate['perPage'] = $request->query->get('perPage', $this->defaultPerPage);
+
+        $result = $this->userRepository->showUsersPaginated($paginate);
+
+        $users = $this->userRepository->findAll();
+
+        return $this->render('Admin/allUsers.html.twig', [
+            'users' => $users
+//            [max(10)]
+//            'paginate' => $paginate,
+//            'result' => $result
+        ]);
+    }
+
+    /**
+     * @Route("admin/user/{id}", name="user_update", requirements={"id"="\d+"})
+     */
+    public function updateUser(Request $request, int $id): Response
+    {
+        $user = $this->userRepository->findOneBy(['id' => $id]);
+        if (null === $user) {
+            $this->addFlash(
+                'error',
+                'User does not exist!'
+            );
+
+            return new Response('User does not exist', Response::HTTP_NOT_FOUND);
+        }
+
+        $form = $this->createForm(UpdateUserType::class)->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $firstName = $form->get('firstName')->getData();
+            $lastName = $form->get('lastName')->getData();
+            $email = $form->get('email')->getData();
+            $phoneNumber = $form->get('phoneNumber')->getData();
+
+            if ($user !== null) {
+                $user->firstName = $firstName;
+                $user->lastName = $lastName;
+                $user->email = $email;
+                $user->setPhoneNumber($phoneNumber);
+
+                $this->entityManager->persist($user);
+                $this->entityManager->flush();
+
+                $this->addFlash(
+                    'success',
+                    'User updated successfully!'
+                );
+
+                return $this->redirectToRoute('admin_users', ['user' => $user]);
+            }
+        }
+
+        return $this->render('Admin/updateUser.html.twig', [
+            'form' => $form->createView(),
+            'user' => $user
+        ]);
+    }
+
+    /**
+     * @Route("admin/user/delete/{id}", name="user_delete")
+     */
+    public function softDeleteAnUser(int $id): Response
+    {
+        $userToDelete = $this->userRepository->findOneBy(['id' => $id]);
+        if (null === $userToDelete) {
+            $this->addFlash(
+                'error',
+                'User does not exist!'
+            );
+
+            return new Response('User does not exist', Response::HTTP_NOT_FOUND);
```

Comment:
same, you are in the admin, you should add a flash message and keep the user on the users page